### PR TITLE
Summary date

### PR DESCRIPTION
**Pull request by joakim-hove**
Support the DATE summary keyword which will expand to 'DAY', 'MONTH' 'YEAR' in the output summary file.

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

File: src/opm/output/eclipse/Summary.cpp [hunk 1]
```diff
@@ -1724,6 +1724,69 @@ namespace Evaluator {
...
+            : saveKey_(std::move(saveKey))
+        {}
+
+        void update(const std::size_t          sim_step,
+                    const double            /* stepSize*/,
+                    const InputData&           input,
+                    const SimulatorResults& /* simRes */,
+                    Opm::SummaryState&         st) const override
+        {
+            auto sim_time = Opm::TimeStampUTC( input.sched.simTime(sim_step + 1) );
```
Comment (bska) [inline src/opm/output/eclipse/Summary.cpp:1740]: That doesn't look completely right to me.  If there are multiple timesteps within a single report step, then all of those will be attributed to the same day.  That's going to be confusing if the report steps have very long duration (6 months, to take an extreme example).

Comment (joakim-hove) [reply src/opm/output/eclipse/Summary.cpp:1740]: You are absolutely right - thank you!

Comment (joakim-hove) [reply src/opm/output/eclipse/Summary.cpp]: Fixed

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
OK - the failures are because the updated case has three new summary vectors - `DAY`, `MONTH` and `YEAR`. I will initiate an `update_data` process

**Comment by joakim-hove** [comment]
jenkins build this update_data please

**Comment by bska** [comment]
> the failures are because the updated case has three new summary vectors

Mostly yes, but the [test_Summary](https://ci.opm-project.org/job/opm-common-PR-builder/1804/testReport/junit/(root)/mpi/Summary/) failure does not appear to be only because of the new summary vectors.

**Comment by joakim-hove** [comment]
> failure does not appear to be only because of the new summary vectors.

You are right - as usual; but now!

**Comment by joakim-hove** [comment]
jenkins build this update_data please

**Comment by joakim-hove** [comment]
jenkins build this opm-tests=280 please

**Comment by joakim-hove** [comment]
jenkins build this update_data please

**Comment by bska** [comment]
jenkins build this opm-tests=280 please

**Pull request merged**